### PR TITLE
resolve conflicts for dplyr/tidyr funcs

### PR DESCRIPTION
**Pull request by chilampoon**
Rewrote Roxygen docs for `dplyr_methods.R` and `tidyr_methods.R` with the same (updated) format in `tidySCE` & `tidyseurat`.

Now the conflicts when loading `tidyomics` become

```R
> library(tidyomics)
── Attaching core tidyomics packages ────────────────────────────────────────────────────────── tidyomics 0.1.1 ──
✔ dplyr                    1.1.2      ✔ tidyr                    1.3.0 
✔ ggplot2                  3.4.3      ✔ tidyseurat               0.7.2 
✔ nullranges               1.6.2      ✔ tidySingleCellExperiment 1.11.6
✔ plyranges                1.19.0     ✔ tidySummarizedExperiment 1.10.0
✔ tidybulk                 1.11.5     
── Conflicts ──────────────────────────────────────────────────────────────────────────── tidyomics_conflicts() ──
✖ plyranges::between()               masks dplyr::between()
✖ ttservice::bind_cols()             masks tidySummarizedExperiment::bind_cols(), dplyr::bind_cols()
✖ ttservice::bind_rows()             masks tidySummarizedExperiment::bind_rows(), dplyr::bind_rows()
✖ tidySummarizedExperiment::count()  masks matrixStats::count(), dplyr::count()
✖ plyranges::filter()                masks tidySummarizedExperiment::filter(), dplyr::filter(), stats::filter()
✖ plyranges::n()                     masks dplyr::n()
✖ plyranges::n_distinct()            masks dplyr::n_distinct()
✖ tidySummarizedExperiment::rename() masks S4Vectors::rename(), dplyr::rename()
✖ plyranges::slice()                 masks tidySummarizedExperiment::slice(), IRanges::slice(), dplyr::slice()
ℹ Use the conflicted package to force all conflicts to become errors
```

@stemangiola do you want me to update `ttservice` & `tidySE` as well..?

### CODE REVIEW

**Comment by stemangiola** [comment]
thanks @chilampoon ! some tests are failing. 

Feel free to fix 'ttservice' as well.

**Comment by chilampoon** [comment]
Locally no error now

```r
0 errors ✔ | 1 warning ✖ | 5 notes ✖
Error: R CMD check found WARNINGs
Execution halted
```

For the deseq2 loading I ending adding those detect & install lines otherwise here

```r
test_that("DESeq2 differential trancript abundance - no object",{

	res_deseq2 =
		test_deseq2_df |>
		DESeq2::DESeq() |>
		DESeq2::results()
```
`DESeq2::` causes errors

File: R/zzz.R [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
+#     ", pkgname, " version ", version, "
+#     If you use TIDYBULK in published research, please cite:
+#     
+#     Mangiola et al. tidybulk: an R tidy framework for modular 
+#     transcriptomic data analysis. Genome Biology 2021.
+#     
+#     This message can be suppressed by:
+#       suppressPackageStartupMessages(library(tidybulk))
+#     ========================================
+#   ")
```
Comment (stemangiola) [inline R/zzz.R:15]: We can keep this

**Review by stemangiola — CHANGES_REQUESTED** [review]

**Comment by stemangiola** [comment]
@chilampoon please have a look to the failed checks.

**Comment by chilampoon** [comment]
It's 0 errors and 0 warnings on my local machine now, I don't know why there was a warning `Warning: Warning: declared S3 method 'filter.tidybulk' not found` in github checks...

**Comment by chilampoon** [comment]
> > It's 0 errors and 0 warnings on my local machine now, I don't know why there was a warning `Warning: Warning: declared S3 method 'filter.tidybulk' not found` in github checks...
> 
> 
> 
> Interesting, @HelenaLC any idea about this? @chilampoon did you follow 100% the `tidySCE` example. The solution must be in some difference between the two. 

Currently the tests are failing because of dependencies, some packages are required to install to finish R CMC CHECK. I think it's better to revise and modify all documentations in the package. I can do it if this task hasn't been assigned. Also if @HelenaLC could review a little that would be helpful, thanks!

**Comment by stemangiola** [comment]
> Currently the tests are failing because of dependencies, some packages are required to install to finish R CMC CHECK.

Apologies, I tried a newer .github action and screwed the CHECKS, I reverted back.

File: tests/testthat/test-bulk_methods.R [hunk 2]
```diff
@@ -851,8 +851,13 @@ test_that("differential trancript abundance - random effects",{
...
 test_that("test prefix",{
 
-	library(DESeq2)
   library(stringr)
+  
+  if (find.package("DESeq2", quiet = TRUE) %>% length %>% equals(0)) {
+    if (!requireNamespace("BiocManager", quietly = TRUE))
+      install.packages("BiocManager", repos = "https://cloud.r-project.org")
+    BiocManager::install("DESeq2", ask = FALSE)
+  }
```
Comment (stemangiola) [inline tests/testthat/test-bulk_methods.R:860]: I will add this to my PR, see we keep 

- drop DESeq2 dependency, and
- resolve conflicts

independent.

**Review by stemangiola — CHANGES_REQUESTED** [review]

**Comment by stemangiola** [comment]
I forgot at what stage we are with this

**Comment by chilampoon** [comment]
> I forgot at what stage we are with this

hi @stemangiola I wonder if anyone is working on the doc/description refactorization or something for tidybulk, if not then I'll probably do it

**Comment by stemangiola** [comment]
(

can you please provide this info for the authorship, if you haven't done already?

<google-sheets-html-origin><style type="text/css"></style>

Email | Institution | First Name | Middle Name(s)/Initial(s) | Last Name | Suffix | Corresponding Author | Home Page URL | Collaborative Group/Consortium | ORCiD
-- | -- | -- | -- | -- | -- | -- | -- | -- | --


)

**Comment by stemangiola** [comment]
> > I forgot at what stage we are with this
> 
> hi @stemangiola I wonder if anyone is working on the doc/description refactorization or something for tidybulk, if not then I'll probably do it

Great, let's first fix the github actions here, and then we merge this PR. We can define the next step after that. I believe the documentation has not been updated yet. Could you please inquire about the existing dedicated issue?

**Comment by chilampoon** [comment]
the issue is still the filter.tidybulk not found warning, which causes the check failure

```
❯ checking S3 generic/method consistency ... WARNING
Warning:   Warning: declared S3 method 'filter.tidybulk' not found
  See section ‘Generic functions and methods’ in the ‘Writing R
  Extensions’ manual.

❯ checking dependencies in R code ... NOTE
  Namespace in Imports field not imported from: ‘pkgconfig’
    All declared Imports should be used.
  Unexported objects imported by ':::' calls:
    ‘glmmSeq:::glmmTMBcore’ ‘glmmSeq:::hyp_matrix’ ‘glmmSeq:::lmer_wald’
    ‘glmmSeq:::organiseStats’
    See the note in ?`:::` about the use of this operator.
```
I don't have this warning locally, I guess it's because of some dependencies problems, plus there are tons of other notes/problems in the report, so I thought it's better to check other scripts as well (not just the dplyr/tidyr functions.R) to resolve the package conflicts..

**Comment by stemangiola** [comment]
> the issue is still the filter.tidybulk not found warning, which causes the check failure
> 
> ```
> ❯ checking S3 generic/method consistency ... WARNING
> Warning:   Warning: declared S3 method 'filter.tidybulk' not found
>   See section ‘Generic functions and methods’ in the ‘Writing R
>   Extensions’ manual.
> 
> ❯ checking dependencies in R code ... NOTE
>   Namespace in Imports field not imported from: ‘pkgconfig’
>     All declared Imports should be used.
>   Unexported objects imported by ':::' calls:
>     ‘glmmSeq:::glmmTMBcore’ ‘glmmSeq:::hyp_matrix’ ‘glmmSeq:::lmer_wald’
>     ‘glmmSeq:::organiseStats’
>     See the note in ?`:::` about the use of this operator.
> ```
> 
> I don't have this warning locally, I guess it's because of some dependencies problems, plus there are tons of other notes/problems in the report, so I thought it's better to check other scripts as well (not just the dplyr/tidyr functions.R) to resolve the package conflicts..

I see, so you strategy would be to also integrate the refactoring a larger part of the documentation, with the hope that this warning would go away.

I think it is worth, although I am pretty sure this warning is due to an isolated cause. Did you double-check that your documentation for filter mirrors 100% the one from `SingleCellExperiment`?

**Comment by chilampoon** [comment]
yes will do it step by step, also most of the SingleCellExperiment documents were refactored

**Comment by stemangiola** [comment]
> yes will do it step by step, also most of the SingleCellExperiment documents were refactored

Great, please inform the others on your plans here

https://github.com/stemangiola/tidybulk/issues/282

we don't want to duplicate efforts. If the other have not started yet (re they are unresponsive, feel free to take action).

**Pull request closed without merging**